src/main/java/org/reaktivity/nukleus/tls/internal/stream/ClientStreamFactory.java

```diff
@@ -449,22 +446,32 @@ private void handleThrottle(
         private void handleWindow(
             final WindowFW window)
         {
-            final int networkWindowBytesDelta = window.update();
-            final int networkWindowFramesDelta = window.frames();
 
-            final int applicationWindowBytesDelta = networkWindowBytesDelta + applicationWindowBytesAdjustment;
-            final int applicationWindowFramesDelta = networkWindowFramesDelta + applicationWindowFramesAdjustment;
+            final int networkWindowCredit = window.credit();
+            final int networkWindowPadding = window.padding();
 
-            applicationWindowBytes += Math.max(applicationWindowBytesDelta, 0);
-            applicationWindowBytesAdjustment = Math.min(applicationWindowBytesDelta, 0);
+            final int applicationWindowCredit = networkWindowCredit + applicationWindowBudgetAdjustment;
 
-            applicationWindowFrames += Math.max(applicationWindowFramesDelta, 0);
-            applicationWindowFramesAdjustment = Math.min(applicationWindowFramesDelta, 0);
+            applicationWindowBudget += Math.max(applicationWindowCredit, 0);
+            applicationWindowBudgetAdjustment = Math.min(applicationWindowCredit, 0);
 
-            if (applicationWindowBytesDelta > 0 || applicationWindowFramesDelta > 0)
+            if (applicationWindowBudget > applicationWindowBudgetMax)
+            {
+                applicationWindowBudgetMax = applicationWindowBudget;
+
+                final int tlsMaxRecordSize = tlsEngine.getSession().getPacketBufferSize();
+                final int tlsMaxRecordCount = Math.max(
+                        (int) Math.ceil((double) applicationWindowBudgetMax / tlsMaxRecordSize), 1);
+
```

Comment:
Not sure if `Math.ceil(...)` is a bit expensive for `double`. Is there any bit shift equivalent operation that would get the same benefit with a cheaper calculation?